### DISCUSSION TITLE
A wee convenience for signals?

### DISCUSSION THREAD

**Discussion by pragdave** [solidjs/solid#1143] [Ideas]
What do y'all think of adding `.get` as a synonym for `this[0]` and `.set` as a synonym for `this[1]` in the return value of `createSignal`?

For example, I currently write:

~~~ javascript
  const psw = createSignal(0)

  dispatcher.register(ET.tPswUpdated, (ev: Event.EmulatorEvent) => {
    psw[1](ev.psw)
  })
~~~

but it would be (marginally) nicer to write

~~~ javascript
  const psw = createSignal(0)
  . . .
    psw.set(ev.psw)
~~~

**Comment by atk** [comment]
Using a tuple is the smallest convenient way to return two items at once. You can write your own abstraction over createSignal:

```js
const createSignalObj = (...args) => {
  const [get, set] = createSignal(...args);
  return { get, set };
};
```

**Comment by msteen** [comment]
If you want the latter syntax, this is already done internally in the store part of Solid:

https://github.com/solidjs/solid/blob/17763fad87af3c34d3ab2eb9df5af40123b11d4d/packages/solid/store/src/store.ts#L122-L136

So you could do something like this (to make your marginally nicer variant work):

```ts
import { Accessor, createSignal, Setter,  } from 'solid-js';
import { SignalOptions } from 'solid-js/types/reactive/signal';

function createSignalCombined<T>(value: T, options: SignalOptions<T>) {
    const [get, set] = createSignal(value, options);
    (get as Accessor<any> & { set: Setter<T> }).set = set;
    return get as Accessor<any> & { set: Setter<T> };
}
```
